### ISSUE TITLE
permutation! requires child parsers results to implement PartialEq

### ISSUE THREAD

**Issue by nomad010** [Geal/nom#724]
The permutation macro requires that the child arguments are PartialEq.

## Prerequisites

- Rust version : rustc 1.26.0-nightly (2789b067d 2018-03-06)
- nom version : "4.0.0-beta2"
- nom compilation features used:

## Test case

The following code doesn't compile as NoPartialEq does not implement PartialEq

```
#[macro_use]
extern crate nom;

struct NoPartialEq {
    value: i32
}

named!(parse_metadata<&str, i32>,
    do_parse!(
        metadata: permutation!(
            map!(tag!("hello"), |_| NoPartialEq { value: 1 }),
            map!(tag!("world"), |_| NoPartialEq { value: 2 })
        ) >>
        (metadata.0.value + metadata.1.value)
    )
);

fn main() {
    parse_metadata("helloworld");
}

```

**Comment by Geal** [comment]
hello,
this should be fixed in 77405b9

**Comment by nomad010** [comment]
Hi there, thanks for your speedy response!

I made the exact same change to my local copy of nom and have been testing it since last night with no problems. I will swap over to your commit instead and start testing that instead.
